src/type/validate.js

```diff
@@ -410,19 +429,53 @@ function validateObjectImplementsInterface(
     }
 
     // Assert additional arguments must not be required.
-    for (const objectArg of objectField.args) {
-      const argName = objectArg.name;
-      const ifaceArg = find(ifaceField.args, arg => arg.name === argName);
-      if (!ifaceArg && isRequiredArgument(objectArg)) {
+    for (const implementingArg of implementingField.args) {
+      const argName = implementingArg.name;
+      const implementedArg = find(
+        implementedField.args,
+        arg => arg.name === argName,
+      );
+      if (!implementedArg && isRequiredArgument(implementingArg)) {
         context.reportError(
-          `Object field ${object.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${iface.name}.${fieldName}.`,
-          [objectArg.astNode, ifaceField.astNode],
+          `Object field ${implementing.name}.${fieldName} includes required argument ${argName} that is missing from the Interface field ${implemented.name}.${fieldName}.`,
+          [implementingArg.astNode, implementedField.astNode],
         );
       }
     }
   }
 }
 
+function validateImplementsAncestors(
+  context: SchemaValidationContext,
+  implementing: GraphQLObjectType | GraphQLInterfaceType,
+  implemented: GraphQLInterfaceType,
+): void {
+  const implementingInterfaces = implementing.getInterfaces();
+  implemented.getInterfaces().forEach(transitive => {
+    if (!implementingInterfaces.includes(transitive)) {
+      if (transitive === implementing) {
```

Comment:
This check is independent of the previous one so why did you nested it or am I missing something here?

Reply:
This is simply to switch which error message is generated (is the interface missing from the list of implemented interfaces due to a circular reference, or because it was never declared). I'm changing this to a ternary to make this logic more clear.